src/utils/pxls/template_manager.py

```diff
@@ -288,6 +290,30 @@ def to_chunks(arr, nrows, ncols):
         )
         return coords_in_canvas
 
+    # From pycharity
+    # https://github.com/Seon82/pyCharity/blob/5eeb48df7990e096da190807714bcd634f806021/src/handlers/pxls/template.py#L38
+    def crop_to_canvas(self, canvas=None) -> tuple[np.ndarray, int, int]:
+        """
+        Crop a numpy array to the canvas boundaries.
+        :return: array, x, y -> array is the cropped array, and x and y the new ox and oy values.
+        """
+        if canvas is None:
+            canvas = stats.board_array
+        min_x = 0 if self.ox > 0 else -self.ox
+        min_y = 0 if self.oy > 0 else -self.oy
+        array = self.palettized_array[min_y:, min_x:]
+        x, y = max(0, self.ox), max(0, self.oy)
+        if y + array.shape[0] > canvas.shape[1]:
+            height = canvas.shape[1] - y
+        else:
+            height = array.shape[0]
+        if x + array.shape[1] > canvas.shape[0]:
+            width = canvas.shape[0] - x
+        else:
+            width = array.shape[1]
+        array = array[:height, :width]
+        return array, x, y
```

Comment:
I think there are some errors in the indexes here (it would explain why the templates get cut below y = 1729 which is the canvas width).
```suggestion
    def crop_to_canvas(self, canvas=None) -> tuple[np.ndarray, int, int]:
        """
        Crop a numpy array to the canvas boundaries.
        :return: array, x, y -> array is the cropped array, and x and y the new ox and oy values.
        """
        if canvas is None:
            canvas = stats.board_array
        min_x = 0 if self.ox > 0 else -self.ox
        min_y = 0 if self.oy > 0 else -self.oy
        array = self.palettized_array[min_y:, min_x:]
        x, y = max(0, self.ox), max(0, self.oy)
        if y + array.shape[0] > canvas.shape[0]:
            height = canvas.shape[0] - y
        else:
            height = array.shape[0]
        if x + array.shape[1] > canvas.shape[1]:
            width = canvas.shape[1] - x
        else:
            width = array.shape[1]
        array = array[:height, :width]
        return array, x, y
```